### PR TITLE
Output correct barcodes to Preparation Sheet

### LINKED ISSUE #299: Output correct barcodes to Preparation Sheet
Barcodes in the mapping file (i.e., prep information sheet) are not correct--i.e., do not match those in the preparation sheet generated by wet lab via existing process--and do not seem to appear in EMP barcode list at all.
Raised in amplicon walkthrough 07/30/2018

### PR DESCRIPTION
**Pull request by AmandaBirmingham**
fixes #299 Output correct barcodes to Preparation Sheet
adds instructions for when to run populate_test_db.sql and when to run populate_prod_db.sql to README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,8 @@ lab manager for plate maps and sequence flows
 Labman relies on the Qiita database. You will need first to install Qiita in
 a different environment (Qiita is Python 2 only, while labman is Python 3) and
 create the Qiita database using the [Qiita installation instructions]
-(https://github.com/biocore/qiita/blob/master/INSTALL.md).
+(https://github.com/biocore/qiita/blob/master/INSTALL.md).  The instructions 
+below assume the Qiita PostgreSQL database is named `qiita-test`.
```
[the remark this answers is not in the document]
Comment (AmandaBirmingham) [reply README.md]: Good catch!

File: README.md [hunk 2]
```diff
@@ -59,14 +60,27 @@ Qiita configuration (for testing purposes):
...
+
+```bash
+psql -d qiita_test -f labman/db/support_files/populate_test_db.sql
+```
+
+to set up the database to support running the unit tests.  Alternately, if 
+creating a production or production-like environment, run:
+
+```bash
+psql -d qiita_test -f labman/db/support_files/populate_prod_db.sql
```
[the remark this answers is not in the document]
Comment (AmandaBirmingham) [reply README.md]: Confusingly enough, yes :)  'qiita_test' is the default name of the database created when one runs a fresh qiita install (see qiita/qiita_core/support_files/config_test.cfg ) and thus has nothing to do with labman or its test-vs-production status per se.  I've added more detail to the discussion of the database name to hopefully make this clearer.

**Comment by AmandaBirmingham** [comment]
Re how many tests are affected by the barcodes:  .... yeah.  Labman doesn't itself use primers/barcodes for anything but rather just tracks which one has been added to each well of each plate.  For shotgun studies (that use i5/i7), that barcode info is put in the sample sheets, but for amplicon studies the only place labman puts barcodes is into the prep info file.

I suppose one could argue that the tests should be explicitly be exercising the barcode-to-well tracking; for example, when I had to rewrite the i5/i7 assignment algorithm used by LibraryPrepShotgunProcess.create, I expanded the test for that method with explicit checks of the primer assignment made for every well.  However, the general idiom in this test suite is to check that the primer assigned to well A1 is correct (e.g., look at `plate_layout[0]0].composition.primer_composition.primer_set_composition.barcode`) and assume all else will also be.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (wasade) [reply README.md]: Ah :) Thank you for adding the additional clarification

**Comment by wasade** [comment]
Ah, I see, okay. I guess we bow to the glory of the one true well and its infinite correctness.... 

I think we should at this stage defer an expansion of the test suite unless we identify a problem warranting it. We should be able to detect unusual barcode-to-well issues at the end of live testing as the produced preparation information file would be bonkers, so I don't think we're incurring risk by deferring.

File: scripts/generate_16s_18s_its_primer_sql [hunk 3]
```diff
@@ -1561,10 +2333,10 @@ ATATAAGGCCCA	CGGCTGCGTTCTTCATCGATGC	8	H12
 """
 
 # from records created by db_patch_manual.sql
-primer_set_ids_by_name = {"EMP 18S": 3, "EMP ITS": 4}
+primer_set_ids_by_name = {"EMP 16S": 1}  # {"EMP 18S": 3, "EMP ITS": 4}
```
Comment (antgonza) [inline scripts/generate_16s_18s_its_primer_sql]: do we really need these comments?

File: scripts/generate_16s_18s_its_primer_sql [hunk 4]
```diff
@@ -1644,8 +2416,9 @@ def generate_insert_values(last_plate_id, last_container_id,
 
 
 if __name__ == "__main__":
+    # use last_plate_id=10, last_container_id=1536,  for 18S + ITS generation
```
Comment (antgonza) [inline scripts/generate_16s_18s_its_primer_sql]: Does this mean that we are not adding 18S + ITS or that we are adding 16S + 18S + ITS? It's confusing looking at the changes and more if we leave this comment there ...

**Pull request merged**